src/test/scala/sigmastate/lang/BinderTest.scala

```diff
@@ -0,0 +1,149 @@
+package sigmastate.lang
+
+import org.scalatest.{PropSpec, Matchers}
+import org.scalatest.prop.PropertyChecks
+import sigmastate._
+import sigmastate.Values._
+import sigmastate.lang.Terms._
+import sigmastate.utxo._
+
+class BinderTest extends PropSpec with PropertyChecks with Matchers {
+
+  def bind(env: Map[String, Any], x: String): SValue = {
+    try {
+      val ast = SigmaParser(x).get.value
+      val binder = new SigmaBinder(env)
+      binder.bind(ast)
+    } catch {
+      case e: Exception =>
+        SigmaParser.logged.foreach(println)
+        throw e
+    }
+  }
+
+  def BoolIdent(name: String): Value[SBoolean.type] = Ident(name).asValue[SBoolean.type]
+  def IntIdent(name: String): Value[SInt.type] = Ident(name).asValue[SInt.type]
+
+  val EV: Map[String, Any] = Map()
+
+  val env = Map("x" -> 10, "y" -> 11, "c1" -> true, "c2" -> false, "arr" -> Array[Byte](1, 2))
+
+  property("simple expressions") {
+    bind(env, "x") shouldBe IntConstant(10)
+    bind(env, "x+y") shouldBe Plus(10, 11)
+    bind(env, "c1 && c2") shouldBe AND(TrueLeaf, FalseLeaf)
+    bind(env, "arr") shouldBe ByteArrayConstant(Array(1, 2))
+    bind(env, "HEIGHT + 1") shouldBe Plus(Height, 1)
+    bind(env, "INPUTS.size > 1") shouldBe GT(SizeOf(Inputs), 1)
+  }
+
+  property("predefined functions") {
+    bind(env, "all(Array(c1, c2))") shouldBe
+        AND(ConcreteCollection(Vector(TrueLeaf, FalseLeaf)))
+  }
+
+    property("let constructs") {
+      bind(env, "{let X = 10; X > 2}") shouldBe
+        Block(Let("X", SInt, IntConstant(10)), GT(IntIdent("X"), 2))
```

Comment:
previously, byte-length id was used to denote user-provided values. the reason for that is efficient serialization (we need to fight for each byte in the blockchain) of identities. I think we should convert string identifiers to bytes